js/exporter/exceljs/export_pivot_grid.js

```diff
@@ -1,47 +1,238 @@
-import { isDefined, isObject } from '../../core/utils/type';
+import { isDefined, isObject, isFunction } from '../../core/utils/type';
 import { Export } from './export';
-import { noop } from '../../core/utils/common';
+import { getDefaultAlignment } from '../../core/utils/position';
+import { camelize } from '../../core/utils/inflector';
 
-const helpers = {
-    _getWorksheetFrozenState(dataProvider, cellRange) {
-        return { state: 'frozen', xSplit: cellRange.from.column + dataProvider.getFrozenArea().x - 1, ySplit: cellRange.from.row + dataProvider.getFrozenArea().y - 1 };
-    },
+const FIELD_HEADERS_SEPARATOR = ', ';
+
+class PivotGridHelpers {
+    constructor(worksheet, dataProvider, options) {
+        this.worksheet = worksheet;
+        this.dataProvider = dataProvider;
+
+        this.topLeftCell = options.topLeftCell;
+
+        this.mergeColumnFieldValues = options.mergeColumnFieldValues;
+        this.mergeRowFieldValues = options.mergeRowFieldValues;
+
+        this.exportFilterFieldHeaders = options.exportFilterFieldHeaders;
+        this.exportDataFieldHeaders = options.exportDataFieldHeaders;
+        this.exportColumnFieldHeaders = options.exportColumnFieldHeaders;
+        this.exportRowFieldHeaders = options.exportRowFieldHeaders;
+
+        this.filterFieldHeaders = this._tryGetFieldHeaders('filter');
+        this.dataFieldHeaders = this._tryGetFieldHeaders('data');
+        this.columnFieldHeaders = this._tryGetFieldHeaders('column');
+        this.rowFieldHeaders = this._tryGetFieldHeaders('row');
+    }
+
+    _getFirstColumnIndex() {
+        return this.topLeftCell.column;
+    }
+
+    _getWorksheetFrozenState(cellRange) {
+        const { x, y } = this.dataProvider.getFrozenArea();
+
+        return {
+            state: 'frozen',
+            xSplit: cellRange.from.column + x - 1,
+            ySplit: cellRange.from.row + y + this._getFieldHeaderRowsCount() - 1,
+        };
+    }
+
+    _getFieldHeaderRowsCount() {
+        let result = 0;
```

Comment:
```suggestion
        let fieldHeaderRowsCount = 0;
```

We can also write this function the following way, don't u like it?
```javascript
    return 0
        + this._allowExportFilterFieldHeaders()
        + (this._allowExportDataFieldHeaders() || this._allowExportColumnFieldHeaders());
```

Reply:
Yep, in this case it looks good